newIDE/app/src/ExportAndShare/LocalExporters/LocalPreviewLauncher/index.js

```diff
@@ -63,22 +65,68 @@ export default class LocalPreviewLauncher extends React.Component<
     hideMenuBar: true,
     alwaysOnTop: true,
     numberOfWindows: 1,
+    captureOptions: null,
   };
   _networkPreviewSubscriptionChecker: ?SubscriptionCheckerInterface = null;
   _hotReloadSubscriptionChecker: ?SubscriptionCheckerInterface = null;
 
   _openPreviewBrowserWindow = () => {
-    const { previewGamePath, previewBrowserWindowOptions } = this.state;
+    const {
+      previewGamePath,
+      previewBrowserWindowOptions,
+      captureOptions,
+      alwaysOnTop,
+      hideMenuBar,
+      numberOfWindows,
+    } = this.state;
     if (!previewBrowserWindowOptions || !previewGamePath) return;
 
     if (!ipcRenderer) return;
 
     ipcRenderer.invoke('preview-open', {
       previewBrowserWindowOptions,
       previewGameIndexHtmlPath: `file://${previewGamePath}/index.html`,
-      alwaysOnTop: this.state.alwaysOnTop,
-      hideMenuBar: this.state.hideMenuBar,
-      numberOfWindows: this.state.numberOfWindows,
+      alwaysOnTop,
+      hideMenuBar,
+      numberOfWindows,
+      captureOptions,
+    });
+
+    ipcRenderer.removeAllListeners('preview-window-closed');
+    ipcRenderer.on('preview-window-closed', async event => {
+      if (captureOptions) {
+        const { screenshots } = captureOptions;
+        if (!screenshots) return;
+        const screenshotPublicUrls: string[] = screenshots.map(
+          screenshot => screenshot.publicUrl
+        );
+
+        // Check if they have been properly uploaded.
+        const responseUploadedScreenshotPublicUrls: Array<
+          string | null
+        > = await Promise.all(
+          screenshotPublicUrls.map(
+            async (screenshotUrl): Promise<string | null> => {
+              const response = await fetch(screenshotUrl, { method: 'HEAD' });
```

Comment:
Same as before, which makes me thing that this is enough to move this to a helper `const uploadedScreenshotPublicUrls = await uploadScreenshotsFromCaptureOptions(captureOptions);`